bolt-contracts/src/contracts/BoltManager.sol

```diff
@@ -1,52 +1,105 @@
-import {IRegistry} from "@symbiotic/interfaces/common/IRegistry.sol";
-import {IOptInService} from "@symbiotic/interfaces/service/IOptInService.sol";
-import {ISlasher} from "@symbiotic/interfaces/slasher/ISlasher.sol";
-import {IVetoSlasher} from "@symbiotic/interfaces/slasher/IVetoSlasher.sol";
-import {IEntity} from "@symbiotic/interfaces/common/IEntity.sol";
-
-import {MapWithTimeData} from "../lib/MapWithTimeData.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {Time} from "@openzeppelin/contracts/utils/types/Time.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {OperatorMapWithTime} from "../lib/OperatorMapWithTime.sol";
+import {EnumerableMap} from "../lib/EnumerableMap.sol";
 import {IBoltValidators} from "../interfaces/IBoltValidators.sol";
+import {IBoltParameters} from "../interfaces/IBoltParameters.sol";
 import {IBoltMiddleware} from "../interfaces/IBoltMiddleware.sol";
 import {IBoltManager} from "../interfaces/IBoltManager.sol";
 
-contract BoltManager is IBoltManager, Ownable {
+/// @title Bolt Manager
+/// @notice The Bolt Manager contract is responsible for managing operators & restaking middlewares, and is the
+/// entrypoint contract for all Bolt-related queries for off-chain consumers.
+/// @dev This contract is upgradeable using the UUPSProxy pattern. Storage layout remains fixed across upgrades
+/// with the use of storage gaps.
+/// See https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
+contract BoltManager is IBoltManager, OwnableUpgradeable, UUPSUpgradeable {
     using EnumerableSet for EnumerableSet.AddressSet;
-    using EnumerableMap for EnumerableMap.AddressToUintMap;
-    using MapWithTimeData for EnumerableMap.AddressToUintMap;
-    using Subnetwork for address;
+    using EnumerableMap for EnumerableMap.OperatorMap;
+    using OperatorMapWithTime for EnumerableMap.OperatorMap;
 
     // ========= STORAGE =========
 
+    /// @notice Bolt Parameters contract.
+    IBoltParameters public parameters;
+
     /// @notice Validators registry, where validators are registered via their
     /// BLS pubkey and are assigned a sequence number.
     IBoltValidators public validators;
 
+    /// @notice Set of operator addresses that have opted in to Bolt Protocol.
+    EnumerableMap.OperatorMap private operators;
+
     /// @notice Set of restaking protocols supported. Each address corresponds to the
     /// associated Bolt Middleware contract.
     EnumerableSet.AddressSet private restakingProtocols;
 
-    // ========= CONSTRUCTOR =========
+    // --> Storage layout marker: 4 slots
+
+    /// @notice Start timestamp of the first epoch.
+    uint48 public START_TIMESTAMP;
```

Comment:
same comment as above. plus, why isn't START_TIMESTAMP counted in the storage layout here?

Reply:
Because it doesn't take up a full storage slot yet. In case other variables smaller than 256 - 48 need to be added they can be added until the slot is full, after which we need to decrement the gap.

Reply:
It is still occupying the 5th slot though; If other variables smaller than 256 - 48 get added, they will share the 5th slot with it. That's what I understood